### PR TITLE
Remove additional config steps required when running Fabric version on Android

### PR DESCRIPTION
**Pull request by kmagiera**
## Description

This PR simplifies the installation process on Android when using Fabric by removing the step when cpp dependency needs to be manually added to the main app's Android.mk file and also getting rid of the registry call that was ought to be added to ReactNativeHost implementation.

## Changes

The following changes has been made:
1. Added ndk step to the screens library configuration on Android (only when fabric is enabled) + ndk build options
2. Setup main library's Android.mk such that it can see RN's dependencies from ReactAndroid folder
3. Load screens native library directly from ScreensPackage which is loaded thanks to autolinking mechanism
4. Register native components with Fabric's component registry directly in JNI init function such that they are installed when shared library is loaded
4.  Updated instructions how screens can be used with Fabric

## Test code and steps to reproduce

Tested this in the following scenarios:
 - build & run FabricExample
 - build & run Example app
 - create new RN68 project, enable fabric, add screens, build & install

### CODE REVIEW

File: README-Fabric.md [hunk 1]
```diff
@@ -1,28 +1,18 @@
...
-    LOCAL_SHARED_LIBRARIES := libjsi \
-    ...
-    -   libyoga
-    +   libyoga \
-    +   librnscreens_modules 
-    ```
+   - In `ReactNativeHost` in your Application (for react native 0.68 template it is in `android/app/src/main/java/com/YOUR_APP_NAME/newarchitecture/MainApplicationReactNativeHost.java`) add
+   ```diff
+   import com.fabricexample.newarchitecture.modules.MainApplicationTurboModuleManagerDelegate;
+   + import com.swmansion.rnscreens.RNScreensComponentRegistry;
```
Comment (kacperkapusciak) [inline README-Fabric.md:12]:
```suggestion
   + import com.swmansion.rnscreens.RNScreensComponentsRegistry;
```

File: android/build.gradle [hunk 2]
```diff
@@ -49,6 +66,20 @@ android {
...
     }
+    if (isNewArchitectureEnabled()) {
+        externalNativeBuild {
+            ndkBuild {
+                path "src/main/jni/Android.mk"
+            }
+        }
+    }
+    packagingOptions {
+        // For some reason gradle only complains about the duplicated versionf of libreact_render libraries
```
Comment (tomekzaw) [inline android/build.gradle:77]:
```suggestion
        // For some reason gradle only complains about the duplicated version of libreact_render libraries
```

File: README-Fabric.md [hunk 3]
```diff
@@ -1,28 +1,18 @@
...
-    + include $(REACT_ANDROID_DIR)/../../react-native-screens/android/src/main/jni/Android.mk
-    include $(CLEAR_VARS)
-    ...
-    LOCAL_SHARED_LIBRARIES := libjsi \
-    ...
-    -   libyoga
-    +   libyoga \
-    +   librnscreens_modules 
-    ```
+   - In `ReactNativeHost` in your Application (for react native 0.68 template it is in `android/app/src/main/java/com/YOUR_APP_NAME/newarchitecture/MainApplicationReactNativeHost.java`) add
```
Comment (tomekzaw) [inline README-Fabric.md:9]:
```suggestion
   - In `ReactNativeHost` in your Application (for React Native 0.68 template it is in `android/app/src/main/java/com/YOUR_APP_NAME/newarchitecture/MainApplicationReactNativeHost.java`) add
```

File: android/build.gradle (hunk 2 above)
Comment (kacperkapusciak) [inline android/build.gradle:77]:
```suggestion
        // For some reason gradle only complains about the duplicated version of libreact_render libraries
```

File: android/build.gradle [hunk 4]
```diff
@@ -49,6 +66,20 @@ android {
...
+        // while there are more libraries copied in intermediates folder of the lib build directory, we exlude
+        // only the ones that make the build fail (ideally we should only include librnscreens_modules but we
+        // are only allowed to specify exlude patterns)
+        exclude "**/libreact_render*.so"
+    }
     sourceSets.main {
         java {
             if(isNewArchitectureEnabled()){
```
Comment (tomekzaw) [inline android/build.gradle:85]:
```suggestion
            if (isNewArchitectureEnabled()) {
```

File: README-Fabric.md [hunk 5]
```diff
@@ -1,28 +1,9 @@
 # React Native Screens - Fabric version
 
-To use this library with your fabric application, you have to:
+To use this library with your Fabric application, you have to:
+
 1. Add latest react-native-screens
 2. on iOS
-    - Install pods using `RCT_NEW_ARCH_ENABLED=1 pod install` (iOS only)
+   - Install pods using `RCT_NEW_ARCH_ENABLED=1 pod install` – this is a ste
```
Comment (Ubax) [inline README-Fabric.md:7]: Should it be `this is the same as in Fabric app`?

**Review by Ubax — COMMENTED** [review]
Code looks good to me

File: android/src/main/java/com/swmansion/rnscreens/RNScreensPackage.kt [hunk 6]
```diff
@@ -4,10 +4,19 @@ import com.facebook.react.ReactPackage
...
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.uimanager.ViewManager
+import com.facebook.soloader.SoLoader
 
 class RNScreensPackage : ReactPackage {
-    override fun createNativeModules(reactContext: ReactApplicationContext) =
-        emptyList<NativeModule>()
+    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
+        if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
+            // For Fabric, we load c++ naitve library here, this triggers screen's Fabric
```
Comment (tomekzaw) [inline android/src/main/java/com/swmansion/rnscreens/RNScreensPackage.kt:12]: native

File: android/build.gradle (hunk 4 above)
Comment (kacperkapusciak) [reply android/build.gradle]: Fixed in [7dc787f](https://github.com/software-mansion/react-native-screens/pull/1322/commits/7dc787f2a68e833147d3ca1f05b544d848e6c5ed)

File: android/src/main/java/com/swmansion/rnscreens/RNScreensPackage.kt (hunk 6 above)
Comment (kacperkapusciak) [reply android/src/main/java/com/swmansion/rnscreens/RNScreensPackage.kt]: Fixed in [bc3f058](https://github.com/software-mansion/react-native-screens/pull/1322/commits/bc3f0588d91028fd98f5b9eeff12134678002d13)

**Review by Ubax — APPROVED** [review]
Looks good to me 🚀 
Just fix the typo in docs

**Pull request merged**